File: 03-ef/lTumilovichProductEF/lTumilovichProductEF/Category.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace lTumilovichProductEF
{
    class Category
    {
        public Category()
        {
            Products = new List<Product>();
        }
        public int CategoryID { get; set; }
        public string Name { get; set; }
        public List<Product> Products { get; set; }
    }
}

File: 04-hibernate/UTumilovichJPAPractice/src/Category.java
import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int categoryId;
    private String name;

    @OneToMany
    @JoinColumn(name="Category_FK")
    private final List<Product> products = new ArrayList<>();

    public Category() {
    }

    public Category(String name) {
        this.name = name;
    }

    public void addProduct(Product product){
        this.products.add(product);
        product.setCategory(this);
    }

    public boolean categoryProduct(Product product) {
        return products.contains(product);
    }
}
